mlir-tensorrt/compiler/lib/Dialect/TensorRTRuntime/Transforms/BufferizableOpInterfaceImpl.cpp

```diff
@@ -41,6 +41,13 @@ static bool isInMemorySpace(Type memrefType, plan::MemorySpace memType) {
   return space == plan::MemorySpaceAttr::get(type.getContext(), memType);
 }
 
+static bool isScalar(Value *input) {
```

Comment:
Don't pass Value by pointer/reference unless it's an output parameter. 

This unction isn't needed as discussed below. 

Also, `isa` can take multiple template parameters to express a disjunction over possible types e.g. `isa<TensorType, MemRefType>(input.getType())`